contents/scalability/scalable-development.md

```diff
@@ -33,9 +33,40 @@ There are a few common bottlenecks that can be avoided such as database querying
+
+The next thing to consider is what to cache. The rule of thumb is to cache data that is frequently accessed and read more often than it is updated. With this, cache hits would be more often than cache misses and as a result, the time saved from faster accesses and reads outweighs the extra time taken to populate the cache.  
+
+1. Scalability using caches ([resource](http://www.lecloud.net/post/9246290032/scalability-for-dummies-part-3-cache))
+
+1. Scalability best practices ([resource](https://www.infoq.com/articles/ebay-scalability-best-practices))
+
+### Sharding 
+As your traffic increases your data increases as well and as a result, your database gets overloaded. One way to mitigate this is to scale your database by sharding. Sharding is a method of splitting and storing a single logical dataset in multiple databases. More specifically, it is the storing of data horizontally - storing rows of a same table in multiple database nodes instead of storing them in the usual vertical way - storing different tables & columns in a separate database.
+
+Sharding is essential when your dataset becomes too large to store in a single database. It reduces the number of rows in each table and as such improves search performance since the search is done on a smaller table. 
+
+Querying the databases containing only the relevant partitions becomes possible as well. For instance, if a database contains a column for age, then you can partition the rows according to an age group and store them in different databases. Whenever there is a need to access the data of a particular age group, instead of querying the whole database, you just need to query the partition that contains that age group. This allows your database to scale along with your data and traffic growth.
+
+1. How Sharding Works ([resource](https://medium.com/@jeeyoungk/how-sharding-works-b4dec46b3f6))
+
+1. A StackOverflow post on Sharding ([resource](https://stackoverflow.com/questions/992988/what-is-sharding-and-why-is-it-important))
+
+### Go Asynchronous 
+Unlike synchronous operations, which runs sequentially and waits for the previous operation to complete before moving on to the next, asynchronous operations are non-blocking and operate independently of each other instead. As a result other operations do not have to wait for asynchronous operations to finish.
+
+If one component A calls another component B synchronously, they are tightly coupled. As a result, to scale A, you must also scale B. By using asynchronous operations, components are decoupled and can be scaled independently of one another.
```

Comment:
Hard to follow this argument. Do you need to mention `coupling` at all? What do you mean by 'scaling a component'?

Reply:
> What do you mean by 'scaling a component'?

By that, I meant if component A has to be modified to deal with higher traffic to component A only, then component B has to be modified as well since A is tightly coupled to B. If asynchronous programming is done, then they are decoupled and only A has to be modified to deal with the higher traffic instead of modifying a component that is not affected by the traffic growth.

But then again if this concept is more confusing than anything, I should remove it instead?